app/src/main/java/com/github/libretube/ui/adapters/InstancesAdapter.kt

```diff
@@ -0,0 +1,41 @@
+package com.github.libretube.ui.adapters
+
+import android.annotation.SuppressLint
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.github.libretube.api.obj.Instances
+import com.github.libretube.databinding.InstanceRowBinding
+import com.github.libretube.ui.viewholders.InstancesViewHolder
+
+class InstancesAdapter(
+    private val instances: List<Instances>,
+    private val onSelectInstance: (index: Int) -> Unit
+): RecyclerView.Adapter<InstancesViewHolder>() {
```

Comment:
This could be a [`ListAdapter`](https://developer.android.com/reference/androidx/recyclerview/widget/ListAdapter) instead. It has additional methods, e.g. getItem.

Reply:
I don't think this makes sense here since we don't actually use any method like `insertItems()` or any diff comparisons for the instances list. The instances list is only initialized once and never modified after.